sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py

```diff
@@ -148,36 +128,51 @@ class TextTranslationClient(ServiceClientGenerated):
 
     Combinations of endpoint and credential values:
     str + AzureKeyCredential - used custom domain translator endpoint
+    str + AzureKeyCredential + Region - used for global translator endpoint
     str + TokenCredential - used for regional endpoint with token authentication
-    str + TranslatorCredential - used for National Clouds
     None + AzureKeyCredential - used for global translator endpoint with global Translator resource
-    None + Token - general translator endpoint with token authentication
-    None + TranslatorCredential - general translator endpoint with regional Translator resource
-
-    :param endpoint: Supported Text Translation endpoints (protocol and hostname, for example:
-         https://api.cognitive.microsofttranslator.com). Required.
-    :type endpoint: str
-    :param credential: Credential used to authenticate with the Translator service
-    :type credential: Union[AzureKeyCredential , TokenCredential , TranslatorCredential, TranslatorAADCredential]
-    :keyword api_version: Default value is "3.0". Note that overriding this default value may
-     result in unsupported behavior.
-    :paramtype api_version: str
+    None + TokenCredential - general translator endpoint with token authentication
+    None + TokenCredential + Region - general translator endpoint with regional Translator resource
+    :keyword str endpoint: Supported Text Translation endpoints (protocol and hostname, for example:
+    https://api.cognitive.microsofttranslator.com). If not provided, global translator endpoint will be used.
+    :keyword credential: Credential used to authenticate with the Translator service
+    :paramtype credential: Union[AzureKeyCredential, TokenCredential]
+    :keyword str region: Used for National Clouds.
+    :keyword str resource_id: Used with both a TokenCredential combined with a region.
+    :keyword str scopes: Scopes of the credentials.
+    :keyword  str api_version: Default value is "3.0". Note that overriding this default value may
+    result in unsupported behavior.
     """
 
+    @overload
     def __init__(
         self,
         *,
-        credential: Optional[Union[AzureKeyCredential, TokenCredential, TranslatorCredential, TranslatorAADCredential]] = None,
+        credential: Optional[AzureKeyCredential] = None,
+        region: Optional[str] = None,
         endpoint: Optional[str] = None,
         api_version="3.0",
         **kwargs
-    ):
+    ): ...
 
-        set_authentication_policy(credential, kwargs)
-
-        translation_endpoint = get_translation_endpoint(endpoint, api_version)
+    @overload
+    def __init__(
+        self,
+        *,
+        credential: Optional[TokenCredential] = None,
+        region: Optional[str] = None,
+        resource_id: Optional[str] = None,
+        endpoint: Optional[str] = None,
+        scopes: Optional[str] = None,
+        api_version="3.0",
+        **kwargs
+    ): ...
 
+    def __init__(self, **kwargs):
```

Comment:
Can you describe what should be passed in the on-prem scenario? No credential + endpoint? This will help me understand how to fix the pyright error here

Reply:
Also our custom pylint doesn't understand overloads yet, so you'll need to `# pylint: disable=client-accepts-api-version-keyword` on this line

Reply:
Onprem scenario is - no auth and endpoint (like http:\\localhost or any DNS entry users might be using in their setup).

Reply:
just to be clear, on-prem does require the endpoint?

Reply:
yes, it needs and endpoint. Comment to doc added for str + None scenario.

Reply:
Okay, I think we can do better here. I'm thinking something like this:

```python
    @overload
    def __init__(
        self,
        *,
        credential: TokenCredential,
        region: str,
        resource_id: str,
        endpoint: Optional[str] = None,
        scopes: Optional[str] = None,
        api_version="3.0",
        **kwargs
    ): ...

    @overload
    def __init__(
        self,
        *,
        credential: AzureKeyCredential,
        region: Optional[str] = None,
        endpoint: Optional[str] = None,
        api_version="3.0",
        **kwargs
    ): ...

    @overload
    def __init__(
        self,
        *,
        endpoint: str,
        api_version="3.0",
        **kwargs
    ): ...
```

Can you verify that the above matches the service behavior? E.g. should region and resource_id be required when using TokenCredential for auth?

Reply:
Agreed offline and pushed changes.